zero_bin/rpc/src/main.rs

```diff
@@ -56,6 +56,15 @@ impl Cli {
             } => {
                 let checkpoint_block_number =
                     checkpoint_block_number.unwrap_or((start_block - 1).into());
+                if let BlockId::Number(checkpoint_block_number) = checkpoint_block_number {
```

Comment:
@Nashtare 
this check which is in rpc binary, should take care of the cases where if we are passing checkpoint_block_number and is less than start_block or we are not passing it at all